plugins/TriggerPrimitiveMaker.cpp

```diff
@@ -74,7 +74,7 @@ TriggerPrimitiveMaker::do_configure(const nlohmann::json& obj)
       tpset.end_time = tpset.start_time + m_conf.tpset_time_width;
       tpset.seqno = seqno++;
       tpset.type = TPSet::Type::kPayload;
-      tpset.from_detids = { tp.detid };
+      // fixme: set tpset.origin
```

Comment:
Let's pick something for `origin` for the moment. How about:

```cpp
system_type = GeoID::SystemType::kDataSelection;
region_id = 0;
element_id = tp.detid;```

If there's a good way to indicate that this is temporary/provisional, we could use that (set `region_id` to 2^16 - 2???)

In the input files we're using (TPC TPs from ProtoDUNE), detid is set (starting at the lowest bit):

* module number  (2 bits) -- 0
* APA number (8 bits) -- 6
* felix link (6 bits) -- 5 (1st file) and 11 (2nd file)

(ie `(module & 0x3) | ((apa &  0xff) << 2) | ((link & 0x3f) << 10)` if I did the math right)

Reply:
`region_id` is set to the "invalid" value by default.  Maybe best to simply keep that until we understand more globally how best to set these bits.

In the end, I think GeoID bits should be set in configuration and not be set in a data-driven way.  

Instead, the module may (should?) apply some (detector specific?) algorithm to compare its configured GeoID to each input datum and warn+drop the input if it is "out of scope".